keyboards/trojan_pinata/model_b/rev0/rev0.c

```diff
@@ -0,0 +1,589 @@
+    oled_set_cursor(14, 1);
+    oled_write_P(PSTR(" SCR "), led_state.scroll_lock);
+
+    // write model B here in invert text
+    oled_set_cursor(0, 3);
+    oled_write_P(PSTR(" MODEL-B "), true);
+
+    // mod masks - if you don't know what these do, don't worry about them
+    uint8_t mod_state = get_mods();
+    oled_advance_char();
+    oled_advance_char();
+    oled_write_P(PSTR("S"), mod_state & MOD_MASK_SHIFT);
+    oled_advance_char();
+    oled_write_P(PSTR("C"), mod_state & MOD_MASK_CTRL);
+    oled_advance_char();
+    oled_write_P(PSTR("A"), mod_state & MOD_MASK_ALT);
+    oled_advance_char();
+    oled_write_P(PSTR("G"), mod_state & MOD_MASK_GUI);
+    oled_advance_char();
+}
+
+// display logo - used on boot and exception state
+static void logo(void) {
+    oled_write_raw_P(bootscreen, sizeof(bootscreen));
+}
+
+// logo, but also have some signifier for the current layer
+static void logo_with_layer(void) {
+    switch (get_highest_layer(layer_state)) {
+        case 0:
+            oled_write_raw_P(bootscreen, sizeof(bootscreen));
+            break;
+        case 1:
+            oled_write_raw_P(invertlogo, sizeof(invertlogo));
+            break;
+        default:
+            oled_write_raw_P(bootscreen, sizeof(bootscreen));
+            break;
+    }
+}
+
+// run this whenever the oled needs to update
+bool oled_task_kb(void) {
+    if (!oled_task_user()) {
+        return false;
+    }
+    keyboard_config.raw = eeconfig_read_kb(); // read saved config
+    display_mode = keyboard_config.display_mode_saved;
+    switch (display_mode) {
+        case MTNS:
+            mountains();
+            break;
+        case INFO:
+            info();
+            break;
+        case LOGO:
+            logo_with_layer();
+            break;
+        case MEW:
+            mew();
+            break;
+        case CEL:
+            celebi();
+            break;
+        case RAY:
+            rayquaza();
+            break;
+        default:
+            logo();
+    }
+    return false;
+}
+
+
+// render this on boot/reset
+void oled_render_boot(bool bootloader) {
+    oled_clear();
+    logo();
+    oled_render_dirty(true);
+}
+
+// run this on boot/reset
+bool shutdown_kb(bool jump_to_bootloader) {
+    if (!shutdown_user(jump_to_bootloader)) {
+        return false;
+    }
+    return false;
+}
+
+// change depending on if OLED_MOD is pressed
+bool process_record_kb(uint16_t keycode, keyrecord_t *record) {
+    if (!process_record_user(keycode, record)) {
+        return false;
+    }
+    if (keycode == OLED_MOD) {
+        if (record->event.pressed) {
+            oled_clear();
+            if (display_mode == NUM_MODES - 1) {
+                display_mode = 0;
+            }
+            else {
+                display_mode++;
+            }
+            keyboard_config.display_mode_saved = display_mode;
+            eeconfig_update_kb(keyboard_config.raw);  // update config when screen changes
+        }
+    }
```

Comment:
If you only have the one keycode, you could merge the if's.  And there are a few things you ecould do to simplify this. 
```suggestion
    if (keycode == OLED_MOD && record->event.pressed) {
        oled_clear();
        keyboard_config.display_mode_saved++;
        if (keyboard_config.display_mode_saved == NUM_MODES) {
            keyboard_config.display_mode_saved = 0;
        }
        eeconfig_update_kb(keyboard_config.raw);
    }
```